modules/core/src/main/java/org/apache/ignite/internal/binary/builder/BinaryObjectBuilderImpl.java

```diff
@@ -247,7 +244,7 @@ void serializeTo(BinaryWriterEx writer, BinaryBuilderSerializer serializer) {
                 int idx = 0;
 
                 while (reader.position() < rawPos) {
-                    int fieldId = schema.fieldId(idx++);
+                    int fieldId = reader.reader().fieldId(idx++);
```

Comment:
`fieldId` should be moved to BinaryUtils, or encapsulated in other way.

Reply:
Sounds reasonable. Thanks.